### PR TITLE
Add status=progressing for version difference

### PR DESCRIPTION
**Pull request by michaelgugino**
Fixes: https://jira.coreos.com/browse/CLOUD-392

Depends-on: https://github.com/openshift/cluster-autoscaler-operator/pull/63

### CODE REVIEW

File: pkg/operator/status.go [hunk 1]
```diff
@@ -191,3 +246,11 @@ func (r *StatusReporter) CheckMachineAPI() (bool, error) {
 	glog.Infof("machine-api-operator not ready yet")
 	return false, nil
 }
+
+func printOperandVersions(desiredVersions []osconfigv1.OperandVersion) string {
```
Comment (frobware) [inline pkg/operator/status.go]: nit: on `print`. it doesn't actually print, so maybe just `operandVersions()`?

File: pkg/operator/status_test.go [hunk 2]
```diff
@@ -93,3 +93,62 @@ func TestApplyConditions(t *testing.T) {
...
+}
+
+func TestPrintOperandVersions(t *testing.T) {
+	desiredVersions := []osconfigv1.OperandVersion{
+		{
+			Name:    "operator",
+			Version: "2.0",
+		},
+	}
+	expectedString := "operator: 2.0"
```
Comment (frobware) [inline pkg/operator/status_test.go]: Given that `[print]operandVersions()` joins the `desiredVersions` together does it make sense to expand the test cases here to cover that case?

File: pkg/operator/status.go (hunk 1 above)
Comment (michaelgugino) [reply pkg/operator/status.go]: This is copied from machine-api-operator.  I'd rather keep the naming the same for that reason.

File: pkg/operator/status_test.go (hunk 2 above)
Comment (michaelgugino) [reply pkg/operator/status_test.go]: Good idea.

**Comment by smarterclayton** [comment]
Pretty sure this is broken - it hasn't passed a single time

/hold

**Comment by ingvagabund** [comment]
Still looks good
/lgtm

**Pull request merged**